File: course_level1_cpp/src/number_reset_counter_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/set_bool.hpp"

using namespace std::chrono_literals;
using namespace std::placeholders;
     
class ResetCounterClientNode : public rclcpp::Node
{
public:
    ResetCounterClientNode() : Node("reset_counter_client")
    {
        client_ = this->create_client<example_interfaces::srv::SetBool>("reset_counter");
    }
    void callResetCounter(bool data)
    {
        while (!client_->wait_for_service(1s)) {
            RCLCPP_WARN(this->get_logger(), "Waiting for the server...");
        }

        auto request = std::make_shared<example_interfaces::srv::SetBool::Request>();
        request->data = data;

        client_->async_send_request(request, std::bind(&ResetCounterClientNode::callbackCallResetCounter, this, _1));
    }
private:
    void callbackCallResetCounter(rclcpp::Client<example_interfaces::srv::SetBool>::SharedFuture future)
    {
        auto response = future.get();
        RCLCPP_INFO(this->get_logger(), "%s", response->message.c_str());
    }

    rclcpp::Client<example_interfaces::srv::SetBool>::SharedPtr client_;
};
    
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<ResetCounterClientNode>();
    node->callResetCounter(true);
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: course_level1_py/course_level1_py/number_reset_counter_client.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import SetBool
from functools import partial
    
class ResetCounterClientNode(Node):
    def __init__(self):
        super().__init__("reset_counter_client")
        self.client_ = self.create_client(SetBool, "reset_counter")

    def call_reset_counter(self, data):
        while not self.client_.wait_for_service(1.0):
            self.get_logger().warn("Waiting for SetBool Server")
        
        request = SetBool.Request()
        request.data = data

        future = self.client_.call_async(request)
        future.add_done_callback(partial(self.callback_call_reset_counter, request=request))

    def callback_call_reset_counter(self, future, request):
        response = future.result()
        self.get_logger().info("Received Request [" 
            + str(request.data) + "]. " + str(response.message))
    
    
def main(args=None):
    rclpy.init(args=args)
    node = ResetCounterClientNode()
    node.call_reset_counter(True)
    # rclpy.spin(node)
    rclpy.shutdown()
    
    
if __name__ == "__main__":
    main()